app/services/connect_with_oauth_provider.rb

```diff
@@ -23,7 +23,7 @@ def connect
   private
 
   def whitelisted
-    return true if whitelist.empty?
+    return true if whitelist.empty? or whitelist.nil?
```

Comment:
a couple things here. 
- don't use ruby `or` or `and`. they're for perl-style operation ordering, `&&` and `||` are for boolean logic.
- if you call `.empty?` on `nil`, it throws `NoMethodError`, so the second part of this statement would never be evaluated
- this is a great use case for Rails' `.blank?. It'll return true if nil, or if an empty string, array, or hash.